src/hazelcore/semantics/CursorInfo.re

```diff
@@ -90,6 +100,251 @@ type t = {
   uses: option(UsageAnalysis.uses_list),
 };
```

Comment:
include cursor_term in CursorInfo.t and also add a parent_term field roughly like this:

```
type parent_term = 
  | ZExp ...
  | ZPat ...
  | ZTyp ...
  | ZBlock ```

These should be computed in cursor_info.